### PR TITLE
fix keepers migration bug

### PR DESCRIPTION
**Pull request by RyanRHall**

### CODE REVIEW

File: core/store/migrate/migrations/0125_keeper_flatten_schema_version.sql [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    FROM jobs
+    WHERE type = 'keeper'
+);
+
+UPDATE jobs
+SET schema_version = 1
+WHERE type = 'keeper';
+
+-- +goose Down
+UPDATE jobs
```
Comment (begmaroman) [inline core/store/migrate/migrations/0125_keeper_flatten_schema_version.sql:53]: Should we update `dot_dag_source ` to the prev version here?

**Review by begmaroman — COMMENTED** [review]
We also need to update all other places (tests, validation, etc) where we use this schema

File: core/store/migrate/migrations/0125_keeper_flatten_schema_version.sql (hunk 1 above)
Comment (infiloop2) [reply core/store/migrate/migrations/0125_keeper_flatten_schema_version.sql:53]: the prev dot_dag_source depends on the schema_version which was buggy due to a wrong migration

Comment (vla33) [reply core/store/migrate/migrations/0125_keeper_flatten_schema_version.sql:53]: Why do we change the `dot_dag_source`, I remember we set it to be the empty string last time? Isn't it sufficient if we just change the schema version to 1 and keep the `dot_dag_source` empty?

File: core/store/migrate/migrations/0125_keeper_flatten_schema_version.sql [hunk 2]
```diff
@@ -0,0 +1,55 @@
+-- +goose Up
+UPDATE pipeline_specs
+SET dot_dag_source = 'encode_check_upkeep_tx   [type=ethabiencode
```
Comment (vla33) [inline core/store/migrate/migrations/0125_keeper_flatten_schema_version.sql:3]: Why do we change the dot_dag_source, I remember we set it to be the empty string last time? Isn't it sufficient if we just change the schema version to 1 and keep the dot_dag_source empty?

File: core/services/keeper/validate_test.go [hunk 3]
```diff
@@ -54,7 +54,7 @@ func TestValidatedKeeperSpec(t *testing.T) {
 			args: args{
 				tomlString: `
 type            			= "keeper"
-schemaVersion   			= 3
+schemaVersion   			= 1
```
Comment (infiloop2) [inline core/services/keeper/validate_test.go:57]: do we enforce the schemaVersion in validation?

File: core/store/migrate/migrations/0125_keeper_flatten_schema_version.sql (hunk 1 above)
Comment (vla33) [reply core/store/migrate/migrations/0125_keeper_flatten_schema_version.sql:53]: I see, it makes sense then why we don't update it 👍

File: core/services/keeper/validate_test.go (hunk 3 above)
Comment (vla33) [reply core/services/keeper/validate_test.go:57]: No, it's not needed, so we can remove it 😊

**Review by vla33 — APPROVED** [review]

**Pull request merged**